lib/modules/dosomething/dosomething_rogue/dosomething_rogue.module

```diff
@@ -164,3 +157,39 @@ function dosomething_rogue_store_rogue_references($rbid, $fid, $rogue_reportback
       ))
     ->execute();
 }
+
+/**
+ * Insert record that stores reference to the most recent uploaded reportback item in
+ * phoenix and it's corresponding id's in Rogue
+ *
+ * @param object $user
+ * @param array  $values
+ * @param int    $run
+ * @param array  $response
+ * @param object $e
+ *
+ */
+function dosomething_rogue_handle_failure($user, $values, $run, $response = NULL, $e = NULL)
+{
+   if (module_exists('stathat')) {
+      stathat_send_ez_count('drupal - Rogue - reportback failed - count', 1);
+    }
+
+    // Save fail to a db log so we can easily export.
+    db_insert('dosomething_rogue_failed_task_log')
+      ->fields([
+        'drupal_id' => $user->uid,
+        'campaign_id' => $values['nid'],
+        'campaign_run_id' => $run->nid,
```

Comment:
Since this in it's own function now, instead of passing `$run` into this function you can just grab it based on the values array with `$run = dosomething_helpers_get_current_campaign_run_for_user($values['nid']);`